src/test/lib/DatagramTest.cpp

```diff
@@ -245,7 +245,15 @@ QuicTestDatagramSend(
                 while (Client.GetDatagramsSuspectLost() != 1 && ++Tries < 10) {
                     CxPlatSleep(100);
                 }
-                TEST_EQUAL(1, Client.GetDatagramsSuspectLost());
+                // NOTE:
+                // There are two cases of DATAGRAM_SEND_STATE_CHANGED
+                // 1. 1->4->1->2->3
+                //      This is expected state changes. 99.99...% cases
+                // 2. 1->2->5->1->2
+                //      This is unexpected. 0.0...1% cases
+                //      state changes to LOST_SUSPECT before the first datagram acked (line 225)
```

Comment:
Why would this datagram be marked as LOST_SUSPECT in the first place? Isn't that the root cause of the failure?

Reply:
I tend to agree. Why would it be lost?

Reply:
I guess it's possible that more packets could be suspected as lost because of such a low RTT triggering the loss detection logic overly aggressively, in the case where an oversubscribed VM runner gets delayed.

Instead of all these changes, just update the code to make sure `Client.GetDatagramsSuspectLost() >= 1` to be more resilient to additional packet loss. Same for the `while` loop above. And instead of this big comment, just put a comment indicating it's possible for more than one packet to get suspected as lost if the machine is running slow (oversubscribed).

Reply:
It seems to me that it's because the delay ACK timer on the peer fired too late. It was scheduled to be fire at 40.643s (40.618+0.025) but it fired at 40.655s. So, the PTO raced with it and won. If it was TCP, I'd do something to effectively disable delay ACK timeout or send 2 packets to trigger an immediate ACK.

Though, no objection to relaxing the `Client.GetDatagramsSuspectLost() >= 1` check.

Reply:
Code all looks fine. Can you please just fix up the comment above like I mentioned above?